pkg/cmd/admin/prune/images.go

```diff
@@ -79,123 +78,216 @@ func NewCmdPruneImages(f *clientcmd.Factory, parentName, name string, out io.Wri
-				blobPruneFunc = func(registryURL, blob string) error {
-					return nil
-				}
-				manifestPruneFunc = func(registryURL, repo, manifest string) error {
-					return nil
-				}
 			}
 
-			if len(cfg.RegistryUrlOverride) > 0 {
-				originalLayerPruneFunc := layerPruneFunc
-				layerPruneFunc = func(registryURL, repo, layer string) error {
-					return originalLayerPruneFunc(cfg.RegistryUrlOverride, repo, layer)
-				}
-				originalBlobPruneFunc := blobPruneFunc
-				blobPruneFunc = func(registryURL, blob string) error {
-					return originalBlobPruneFunc(cfg.RegistryUrlOverride, blob)
-				}
-				originalManifestPruneFunc := manifestPruneFunc
-				manifestPruneFunc = func(registryURL, repo, manifest string) error {
-					return originalManifestPruneFunc(cfg.RegistryUrlOverride, repo, manifest)
-				}
-			}
-
-			pruner.Run(imagePruneFunc, imageStreamPruneFunc, layerPruneFunc, blobPruneFunc, manifestPruneFunc)
+			err = pruner.Prune(imagePruner, imageStreamPruner, layerPruner, blobPruner, manifestPruner)
+			cmdutil.CheckErr(err)
 		},
 	}
 
 	cmd.Flags().BoolVar(&cfg.Confirm, "confirm", cfg.Confirm, "Specify that image pruning should proceed. Defaults to false, displaying what would be deleted but not actually deleting anything.")
 	cmd.Flags().DurationVar(&cfg.KeepYoungerThan, "keep-younger-than", cfg.KeepYoungerThan, "Specify the minimum age of a build for it to be considered a candidate for pruning.")
-	cmd.Flags().IntVar(&cfg.TagRevisionsToKeep, "keep-tag-revisions", cfg.TagRevisionsToKeep, "Specify the number of image revisions for a tag in an image stream that will be preserved.")
+	cmd.Flags().IntVar(&cfg.KeepTagRevisions, "keep-tag-revisions", cfg.KeepTagRevisions, "Specify the number of image revisions for a tag in an image stream that will be preserved.")
 	cmd.Flags().StringVar(&cfg.CABundle, "certificate-authority", cfg.CABundle, "The path to a certificate authority bundle to use when communicating with the OpenShift-managed registries. Defaults to the certificate authority data from the current user's config file.")
 	cmd.Flags().StringVar(&cfg.RegistryUrlOverride, "registry-url", cfg.RegistryUrlOverride, "The address to use when contacting the registry, instead of using the default value. This is useful if you can't resolve or reach the registry (e.g.; the default is a cluster-internal URL) but you do have an alternative route that works.")
 
 	return cmd
 }
 
+// describingImageStreamPruner prints information about each image stream update.
+// If a delegate exists, its PruneImageStream function is invoked prior to returning.
+type describingImageStreamPruner struct {
+	w             io.Writer
+	delegate      prune.ImageStreamPruner
+	headerPrinted bool
+}
+
+var _ prune.ImageStreamPruner = &describingImageStreamPruner{}
+
+func (p *describingImageStreamPruner) PruneImageStream(stream *imageapi.ImageStream, image *imageapi.Image, updatedTags []string) (*imageapi.ImageStream, error) {
```

Comment:
I assume it's better to put the godoc on the interface where the func is defined?